src/js/view/controls/tizen/tizen-controls.ts

```diff
@@ -0,0 +1,209 @@
+import Title from 'view/title';
+import TizenControlbar from 'view/controls/tizen/tizen-controlbar';
+import DisplayContainer from 'view/controls/display-container';
+import NextUpToolTip from 'view/controls/nextuptooltip';
+import SettingsMenu from 'view/controls/components/menu/settings-menu.js';
+import { addClass } from 'utils/dom';
+import { STATE_PLAYING, STATE_PAUSED } from 'events/events';
+import type ViewModel from 'view/view-model';
+import type { PlayerAPI } from 'types/generic.type';
+
+const Controls = require('../controls').default;
+
+const ACTIVE_TIMEOUT = 5000;
+
+const reasonInteraction = () => {
+    return { reason: 'interaction' };
+};
+
+class TizenControls extends Controls {
+    context: HTMLDocument;
+    playerContainer: HTMLElement;
+    div: HTMLElement | null;
+    backdrop: HTMLElement | null;
+    pauseDisplay: HTMLElement | null;
+    displayContainer: DisplayContainer | null;
+    controlbar: TizenControlbar | null;
+    settingsMenu: SettingsMenu | null;
+    showing: boolean;
+    keydownCallback: ((evt: KeyboardEvent) => void) | null;
+    addControls: any;
+    addBackdrop: any;
+    trigger: any;
+
+    constructor(context: HTMLDocument, playerContainer: HTMLElement) {
+        super(context, playerContainer);
+
+        this.context = context;
+        this.playerContainer = playerContainer;
+        this.div = null;
+        this.backdrop = null;
+        this.pauseDisplay = null;
+        this.displayContainer = null;
+        this.controlbar = null;
+        this.settingsMenu = null;
+        this.showing = false;
+        this.keydownCallback = null;
+    }
+
+    enable(api: PlayerAPI, model: ViewModel): void {
+        addClass(this.playerContainer, 'jw-tizen-app');
+
+        const element = this.context.createElement('div');
+        element.className = 'jw-tizen-controls jw-tizen-reset';
+        this.div = element;
+    
+        const backdrop = this.context.createElement('div');
+        backdrop.className = 'jw-controls-backdrop jw-reset';
+        this.backdrop = backdrop;
+
+        // Pause Display
+        if (!this.pauseDisplay) {
```

Comment:
what are your thoughts on using templating for this similar to how we construct other complex elements? (https://github.com/jwplayer/jwplayer-commercial/blob/master/src/js/templates/dismissButton.js)